Remove PBS and SGE platforms

kinow opened the pull request:
As discussed in today's Autosubmit meeting, SGE and PBS platforms have not been updated nor tested since the AS3 → AS4 update. There are functions that are missing, and they will not work if any user tries them and, besides that, they are lacking tests too.

So it's better for us to remove so users are not misled into using them, and then we can add them back again later when needed, with tests.

**Check List**

- [x] I have read `CONTRIBUTING.md`.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to `pyproject.toml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included in `CHANGELOG.md` if this is a change that can affect users.
- [x] Documentation updated.
- [x] If this is a bug fix, PR should include a link to the issue (e.g. `Closes #1234`).

kinow commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+### 4.1.15 (Not released)
```
@dbeltrankyl I started the blank Changelog for the next release... not sure if you had another idea for this?

kinow commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+
+**Bug fixes:**
+
+- Removed PBS and SGE platforms as they are not working in AS4 #2349
```
@dbeltrankyl what about using just the hash and the issue number? It should work on GH, with the advantage that the link is relative to current project (so if the org/project are renamed or moved within GH that should still work), and they are a bit easier/faster to write :grimacing:

kinow commented:
As agreed in today's meeting. Ready for review!

dbeltrankyl reviewed (approved):
Looks good! I've resolved the conflicts (changelog)

kinow commented:
> Looks good! I've resolved the conflicts (changelog)

Thanks Dani! I updated the branch and squashed commits. Waiting for CICD pipeline then will merge it if everything goes well.
